phalcon/Http/Response.zep

```diff
@@ -546,18 +547,31 @@ class Response implements ResponseInterface, InjectionAwareInterface, EventsAwar
     public function setFileToSend(string filePath, attachmentName = null, attachment = true) -> <ResponseInterface>
     {
         var basePath;
+        var basePathEncoding = "ASCII";
+        var basePathEncoded;
 
         if typeof attachmentName != "string" {
-            let basePath = basename(filePath);
+            let basePath = Fs::basename(filePath);
         } else {
             let basePath = attachmentName;
         }
-
         if attachment {
+            let basePathEncoded = rawurlencode(basePath);
+            // mbstring is a non-default extension
+            if function_exists("mb_detect_encoding") {
+                let basePathEncoding = mb_detect_encoding(
+                    basePath,
+                    mb_detect_order(),
+                    true
+                );
+            }                
             this->setRawHeader("Content-Description: File Transfer");
             this->setRawHeader("Content-Type: application/octet-stream");
-            this->setRawHeader("Content-Disposition: attachment; filename=" . basePath . ";");
+            this->setRawHeader("Content-Disposition: attachment; filename=" . basePathEncoded . ";");
             this->setRawHeader("Content-Transfer-Encoding: binary");
+            if basePathEncoding != "ASCII" {
+                this->setRawHeader("Content-Disposition: attachment; filename=" . basePathEncoded . "; filename*=". strtolower(basePathEncoding) . "''" . basePathEncoded . ";");
+            }
```

Comment:
I would like to avoid duplicate `Content-Disposition` headers. It may cause  an error on Chrome, and an empty page in Firefox. This can be achieved with `Headers::set("Content-Disposition", value)` (instead of using `setRaw`) or:
```zep
if basePathEncoding != "ASCII" {
    // ...
} else {
    this->setRawHeader("Content-Disposition: attachment; filename=" . basePathEncoded . ";");
}
```

Reply:
Based on the comment of @StudioMaX I think this should look like this:
```
if basePathEncoding != "ASCII" {
    // ...
} else {
    this->setRawHeader("Content-Disposition: attachment; filename='" . basePathEncoded . "'");
}
```
Removing trailing ';' and quotes around filename (in case of space)